scanner/nftScanner.py

```diff
@@ -32,71 +45,109 @@ def __init__(self, cf_contract_address, so_contract_address, from_block):
         self.cf_abi_file_path = '../contracts/abi/LagrangeChainlinkData.json'
         # Data NFT with single oracle contract ABI
         self.so_abi_file_path = '../contracts/abi/LagrangeChainlinkDataConsumer.json'
-        self.from_block = from_block
-        self.batch_size = 1000
 
         # DB connection
-        self.mydb = mysql.connector.connect(
-            host="localhost",
-            user=config('DB_USER'),
-            password=config('DB_PASSWORD'),
-            database='nft_data'
-        )
+        self.engine = create_engine('mysql+mysqlconnector://' + config('DB_USER') + ':' + config('DB_PASSWORD') + '@localhost/nft_data')
```

Comment:
The engine should be created when initializing the service. The service should share a connection pool. It is also duplicated with `db = SQLAlchemy(app)` in app.py. Also, the connection loop size should be limited.